lib/wallaby/httpclient.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Wallaby.HTTPClient do
+  @type method :: :post | :get | :delete
+  @type url :: String.t
+  @type params :: map | String.t
+  @type request_opts :: {:encode_json, boolean}
+
+  @doc """
+  Sends a request to the webdriver API and parses the
+  response.
+  """
+  @spec request(method, url, params, [request_opts]) :: {:ok, any}
+                                                      | {:error, :invalid_selector}
+                                                      | {:error, :stale_reference}
+
+  def request(method, url, params \\ %{}, opts \\ [])
+  def request(method, url, params, _opts) when map_size(params) == 0 do
+    make_request(method, url, "")
+  end
+  def request(method, url, params, [{:encode_json, false} | _]) do
+    make_request(method, url, params)
+  end
+  def request(method, url, params, _opts) do
+    make_request(method, url, Poison.encode!(params))
+  end
+
+  defp make_request(method, url, body) do
+    HTTPoison.request(method, url, body, headers(), request_opts())
+    |> handle_response
+    |> case do
+         {:error, :httpoison} ->
+           make_request(method, url, body)
```

Comment:
Yes thats definitely a possibility. We need to only retry a few times before throwing an error. Honestly I'm not sure why chromedriver responses are so flaky. It would be great if it didn't throw errors all the time :/